app/services/moab_storage_service.rb

```diff
@@ -2,6 +2,20 @@
 
 # Responsibility: interactions with Moab storage to support read-only access for ReST API calls
 class MoabStorageService
+  # @param [String] druid
+  # @param [String] category category of desired file ('content', 'metadata', or 'manifest')
+  # @param [String] filename name of the file (path relative to base directory)
+  # @param [String] version of the file (nil = latest)
+  # @return [Pathname] Pathname object containing the full path for the specified file
+  # @raise [Moab::ObjectNotFoundException] if file not found
+  def self.retrieve_file(druid, category, filename, version=nil)
+    raise(ArgumentError, "No filename provided to MoabStorageService.retrieve_file for druid #{druid}") if filename.blank?
+    err_msg = "category arg must be 'content', 'metadata', or 'manifest' (MoabStorageService.retrieve_file for druid #{druid})"
+    raise(ArgumentError, err_msg) unless ['content', 'metadata', 'manifest'].include?(category)
```

Comment:
I wrote this before I wrote the controller method.  Do you think these lines should be removed in favor of the param validation in the objects controller?   Or leave it in both places?

Reply:
I prefer the validation be in the storage service.  I don't like having it duplicated though.

Reply:
Agree. The storage service cares more about the values of those params ultimately.